docs/content/extensions.md

```diff
@@ -0,0 +1,229 @@
+
+```console
+mkdir ocis-extension-workshop && ocis-extension-workshop
+```
+
+Following https://github.com/owncloud/ocis
+
+```console
+git clone https://github.com/owncloud/ocis.git
+cd ocis
+
+TAGS=simple make generate build
+```
+
+*Q: Can you specify which version of phoenix to use?*
+*A: No, the phoenix that is used is compiled into the [assets of ocis-phoenix](https://github.com/owncloud/ocis-phoenix/blob/master/pkg/assets/embed.go) which is currently not automatically updated. We'll see how to use a custom phoenix later.*
+
+`bin/ocis server`
+
+Open the browser at http://localhost:9100
+
+1. You land on the login screen. click login
+2. You are redirected to an idp at http://localhost:9140/oauth2/auth with a login mask. Use `einstein:relativity`to login (one of the three demo users)
+3. You are redirected to http://localhost:9100/#/hello the ocis-hello app
+4. Replace `World` with something else and submit. You should see `Hello %something else%`
+
+*Q: One of the required ports is already in use. Ocis seems to be trying to restart the service over and over. What gives?*
+*A: Using the ocis binary to start the server will case ocis to keep track of the different services and restart them in case they crash.*
+
+## Hacking ocis-hello
+
+go back to the ocis-extension-workshop folder
+
+```console
+cd ..
+```
+
+Following https://github.com/owncloud/ocis-hello
+
+```
+git clone https://github.com/owncloud/ocis-hello.git
+cd ocis-hello
+
+yarn install
+# this actually creates the assets
+yarn build
+
+# this will compile the assets into the binary
+make generate build
+```
+
+Two options:
+1. run only the necessery services from ocis and ocis-hello independently
+2. compile ocis with the updated ocis-hello
+
+### Option 1:
+get a list of ocis services:
+
+```console
+ps ax | grep ocis
+```
+
+Try to kill `ocis hello`
+
+Remember: for now, killing a service will cause ocis to restart it. This is subject to change.
+
+In order to be able to manage the processes ourselves we need to start them independently:
+
+`bin/ocis server` starts the same services as:
+
+```
+bin/ocis micro &
+bin/ocis phoenix &
+bin/ocis hello &
+bin/ocis reva &
+```
+
+Now we can kill the `ocis hello` and use our custom built ocis-hello binary:
+
+```console
+cd ../ocis-hello
+bin/ocis-hello server
+```
+
+## Hacking phoenix (and ocis-phoenix)
+
+Following https://github.com/owncloud/phoenix we are going to build the current phoenix
+
+```
+git clone https://github.com/owncloud/phoenix.git
+cd phoenix
+
+yarn install
+yarn dist
+```
+
+We can tell ocis to use the compiled assets:
+
+Kill `ocis phoenix`, then use the compiled assets when starting phoenix.
+
+```console
+cd ../ocis
+PHOENIX_ASSET_PATH="`pwd`/../phoenix/dist" bin/ocis phoenix
+```
+
+## The ownCloud design system
+
+The owncloud design system contains a set of recommended vue components for phoenix or your own ocis extensions. Use it for a consistent look and feel.
```

Comment:
Maybe `a set of ownCloud vue components...` might be better? This seems more like a collection of some components.